packages/graph/hash_graph/lib/graph/src/datastore/postgres/mod.rs

```diff
@@ -146,169 +150,144 @@ impl PostgresDatabase {
         Ok(())
     }
 
-    /// Inserts a [`DataType`] identified by [`VersionId`], and associated with an [`AccountId`],
-    /// into the database.
+    /// Inserts the specified [`DatabaseType`].
+    ///
+    /// This first checks, if the [`BaseUri`] of the type does not already exist. If not, it will
+    /// insert the [`BaseUri`] and creates a new [`VersionId`] from the contained [`VersionedUri`]
+    /// and inserts the type.
```

Comment:
Isn't that what the first check captures? If something (the `BaseUri`) already exists, we error (`BaseUriAlreadyExists`) out and the user should update instead. Or do you mean we should have a more specific error type for that case

Reply:
To be more specific I think the issue is:

> This first checks if the [`BaseUri`] exists ...

I don't think we should really be describing failure conditions in the opening line of a doc-comment.

We can save them for the ` # Errors` section

Reply:
The opening line is "Inserts the specified [`DatabaseType`].". As mentioned before, this is a comment on an implementation detail, which is only accessible in this module, so this does not even appear in public documentation.

Reply:
What about
```rust
    /// This first extracts the [`BaseUri`] from the [`VersionedUri`] and attempts to insert it into
    /// the database. It will create a new [`VersionId`] for this [`VersionedUri`] and inserts the
    /// entry.
```